### ISSUE TITLE
Neo4j::ActiveNode::Query::QueryProxy bug with .first and .last methods

### ISSUE THREAD

**Issue by kopylovvlad** [neo4jrb/neo4j#1486]
QueryProxy's methods .first and .last always return first element. It makes me sad

Additional information which could be helpful if relevant to your issue:

### Code example (inline, gist, or repo)
```ruby
users = User.all
users.size
# => 12 
users.class 
# => Neo4j::ActiveNode::Query::QueryProxy 
users.first.id
# User 
#  MATCH (n:`User`)
#  RETURN n
#  ORDER BY ID(n)
#  LIMIT {limit_1} | {:limit_1=>1}
# HTTP REQUEST: 3ms POST http://localhost:7575/db/data/transaction/commit (1 bytes)
# => 47 

users.last.id
# User 
#  MATCH (n:`User`)
#  RETURN n
#  ORDER BY ID(n)
#  LIMIT {limit_1} | {:limit_1=>1}
# HTTP REQUEST: 2ms POST http://localhost:7575/db/data/transaction/commit (1 bytes)
# => 47 

users.to_a.first.id
# => 47 
users.to_a.last.id
# => 393 
users[0].id
# => 47 
users[-1].id
# => 393 
```

### Runtime information:

Neo4j database version:
`neo4j` gem version: 9.1.4
`neo4j-core` gem version: 8.1.1
`ruby` verion: 2.4.1

**Comment by thefliik** [comment]
Hmmm, looking at the cypher you included with your issue, it appears the problem is that both `user.last.id` and `user.first.id` are `ORDER BY ID(n)`.

I imagine a fix is updating `user.first.id` to be `ORDER BY ID(n) ASC` and `user.last.id` to be `ORDER BY ID(n) DESC`.

I think I've run into this issue a few times myself, though I never diagnosed what was going on.

**Comment by thefliik** [comment]
Hmm, looks like the `#last()` definition already is suppose to be ordering in DESC. https://github.com/neo4jrb/neo4j/blob/d3a2412a45c551225b24f4a21760183c3f7e561e/lib/neo4j/active_node/query_methods.rb#L19-L22

**Comment by thefliik** [comment]
Ah ok, so while the _active node_ `.last()` query method seems to correctly order DESC, the _query proxy_ `.last()` query method does not appear to do this.

You can check out [QueryProxy#last](https://github.com/neo4jrb/neo4j/blob/master/lib/neo4j/active_node/query/query_proxy_methods.rb#L34-L36) which actually delegates most of its logic to [QueryProxy#first_and_last](https://github.com/neo4jrb/neo4j/blob/master/lib/neo4j/active_node/query/query_proxy_methods.rb#L257-L269). I'm not sure what the `FIRST` and `LAST` constants are _suppose_ to be doing, but when the query proxy object doesn't currently have any `order` clause, those two arguments appear to be ignored by `first_and_last`.

The `first_and_last` method should probably be updated to something like

```
def first_and_last(func, target)
  new_query, pluck_proc = if self.query.clause?(:order)
                            [
                              self.query.with(identity),
                              proc { |var| "#{func}(COLLECT(#{var})) as #{var}" }
                            ]
                          elsif func == LAST
                            [
                              self.order({order_property => :DESC}).limit(1),
                              proc { |var| var }
                            ]
                          else
                            [
                              self.order(order_property).limit(1),
                              proc { |var| var }
                            ]
                          end
  query_with_target(target) do |var|
    final_pluck = pluck_proc.call(var)
    new_query.pluck(final_pluck)
  end.first
end
```

I'm not 100% sure about the `.order()` syntax, but this is probably a pretty easy PR to put together if you're interested in fixing this.

**Comment by kopylovvlad** [comment]
@thefliik  thank you, for your response. I am happy that you know how to fix this bug. 
I created PL to master for fix. Here is it: https://github.com/neo4jrb/neo4j/pull/1494

**Comment by cheerfulstoic** [comment]
#1494 merged and fix released as 9.2.1.  Feel free to comment if something still isn't right
